src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioMutation.cs

```diff
@@ -43,6 +43,33 @@ public void MutabilityTest()
             }
         }
 
+        [Fact]
+        public void MutabilityTest_Chain()
+        {
+            var config = new PowerFxConfig(powerFxFlags: PowerFxFlags.EnableExpressionChaining);            
+            config.AddFunction(new Assert2Function());
+            config.AddFunction(new Set2Function());
+            var engine = new RecalcEngine(config);
+
+            var d = new Dictionary<string, FormulaValue>
+            {
+                ["prop"] = FormulaValue.New(123)
+            };
+
+            var obj = MutableObject.New(d);
+            engine.UpdateVariable("obj", obj);
+
+            var exprs = new string[]
+            {
+                "Assert2(obj.prop, 123); Set2(obj, \"prop\", 456); Assert2(obj.prop, 456)"
```

Comment:
>"Assert2(obj.prop, 123); Set2(obj, \"prop\", 456); Assert2(obj.prop, 456)" [](http://example.com/codeflow?start=16&length=74)

Good test. 
Note expr can now be a string, not a string[].

Reply:
Added one more test in the array

Reply:
Just make this a [Theory] and avoid string[] / for-loop.